File: src/main/org/nlogo/prim/threed/_orbitup.java
package org.nlogo.prim.threed;

import org.nlogo.api.LogoException;
import org.nlogo.nvm.Command;
import org.nlogo.nvm.Syntax;

public final strictfp class _orbitup
    extends Command {
  @Override
  public Syntax syntax() {
    return Syntax.commandSyntax
        (new int[]{Syntax.TYPE_NUMBER}, "O---", true);
  }

  @Override
  public void perform(final org.nlogo.nvm.Context context) throws LogoException {
    double delta = argEvalDoubleValue(context, 0);
    world.observer().orbitUp(delta);
    context.ip = next;
  }
}

File: src/main/org/nlogo/prim/threed/_orbitup.scala
package org.nlogo.prim.threed

import org.nlogo.api.LogoException
import org.nlogo.nvm.{ Command, Context, Syntax }

class _orbitup extends Command {
  override def syntax =
    Syntax.commandSyntax(Array(Syntax.TYPE_NUMBER), "O---", true)
  override def perform(context: Context) {
    world.observer.orbitUp(argEvalDoubleValue(context, 0))
    context.ip = next
  }
}
